WooCommerce/Classes/ViewModels/Order Details/Shipping Labels/BoldableTextView.swift

```diff
@@ -0,0 +1,113 @@
+    var toTextView: Text {
+        if isBold {
+            return Text(content).fontWeight(.bold)
+        } else {
+            return Text(content)
+        }
+    }
+}
+
+private extension String {
+    /// Parses the input text and returns a collection of boldable elements.
+    ///
+    /// - Returns: A collection of boldable elements.
+    func parseBoldableElements() -> [BoldableTextView.Element] {
+        let boldRegionMark = BoldableTextView.boldRegionMark
+        let regex = try! NSRegularExpression(pattern: "\\\(boldRegionMark){1}(.*?)\\\(boldRegionMark){1}")
+        let range = NSRange(location: 0, length: count)
+
+        /// Finds all the ranges that match the regex for content within a bold region.
+        let matches: [NSTextCheckingResult] = regex.matches(in: self, options: [], range: range)
+        let matchingRanges = matches.compactMap { Range<Int>($0.range) }
+
+        guard matchingRanges.isNotEmpty else {
+            return [BoldableTextView.Element(content: self, isBold: false)]
+        }
+
+        var elements: [BoldableTextView.Element] = []
+
+        var startingIndex = 0
+        matchingRanges.enumerated().forEach { index, matchingRange in
+            // Adds an element for the range from the last index and before the matching range, if non-empty.
+            let rangeBefore = startingIndex..<matchingRanges[index].lowerBound
+            if rangeBefore.isEmpty == false {
+                let content = self[rangeBefore]
+                elements.append(BoldableTextView.Element(content: content, isBold: false))
+            }
+
+            // Adds an element for the matching range which should be bold.
+            let matchContent = self[matchingRange]
+            elements.append(BoldableTextView.Element(content: matchContent, isBold: true))
+
+            // Updates the last starting index so that the next range starts after the end of current matching range.
+            startingIndex = matchingRange.upperBound
+
+            // If the matching range is the last one, adds an element for the range after the last matching range, if non-empty.
+            let isLastMatchingRange = matchingRange == matchingRanges.last
+            if isLastMatchingRange {
+                let lastRange = matchingRange.upperBound..<count
+                if lastRange.isEmpty == false {
+                    let content = self[lastRange]
+                    elements.append(BoldableTextView.Element(content: content, isBold: false))
+                }
+            }
+        }
+        return elements
+    }
+
+    /// - Returns: A string subscript based on the given range.
+    subscript(range: Range<Int>) -> String {
+        let startIndex = index(self.startIndex, offsetBy: range.lowerBound)
+        let endIndex = index(self.startIndex, offsetBy: range.upperBound)
+        return String(self[startIndex..<endIndex])
+    }
+}
+
+struct RichTextView_Previews: PreviewProvider {
```

Comment:
I would suggest considering separating previews to a different file.

Reply:
what do you think are the benefits of having the previews in a different file? before this PR, we already have a few custom `UIView`s that we add SwiftUI previews in the same file (like `InAppFeedbackCardViewController`, `PrimarySectionHeaderView`, etc.)

while developing the SwiftUI views, I also found it convenient to have the preview code in the same file to edit preview data & preview without leaving the window. curious to hear other thoughts 🙂